File: algorithms/c#/NormalSort.cs
namespace Algorithms
{
	public static class NormalSort
	{
		public static void Sort(int[] array)
		{
			int min = array[0];
			int max = array[0];
			int[] sortedArray = new int[array.Length];

			for(int i = 0; i < array.Length; i++)
			{
				if(array[i] < min) min = array[i];
				if(array[i] > max) max = array[i];
			}

			for(int i = 0; i < array.Length; i++)
			{
				int index = (int)(((float)(array[i] - min) / (float)(max - min)) * (array.Length - 1));
				sortedArray[index] = array[i];
			}

			for(int i = 0; i < array.Length; i++)
			{
				array[i] = sortedArray[i];
			}
		}
	}
}

File: algorithms/go/normal_sort.go
package algorithms

func NormalSort(slice []int) {
	if(len(slice) < 2) {
		return
	}

	min := slice[0]
	max := slice[0]
	sortedSlice := make([]int, len(slice))

	for i := 0; i < len(slice); i++ {
		if(slice[i] < min) {
			min = slice[i]
		}

		if(slice[i] > max) {
			max = slice[i]
		}
	}

	for i := 0; i < len(slice); i++ {
		index := int((float64(slice[i] - min) / float64(max - min)) * float64(len(slice) - 1));
		sortedSlice[index] = slice[i]
	}

	copy(slice, sortedSlice)
}